### PR TITLE
Fix removing service-account from group

### PR DESCRIPTION
**Pull request by fhlavac**
### Description

Fixed removing service account from group - we were not creating the API request appropriately

[RHCLOUD-32002](https://issues.redhat.com/browse/RHCLOUD-32002)

---

#### Before:
image

#### After:
image

---

### Checklist ☑️
- [x] PR only fixes one issue or story 
- [x] Change reviewed for extraneous code 
- [x] UI best practices adhered to 
- [x] Commits squashed and meaningfully named 
- [x] All PR checks pass locally (build, lint, test, E2E)

##
- [-] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [-] _(Optional) QE: Has been mentioned_
- [-] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [-] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

File: src/helpers/service-account/service-account-helper.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export interface ServiceAccount {
 }
 
 export interface ServiceAccountsPayload {
-  data: ServiceAccount[];
+  data: (ServiceAccount & { clientId: string })[];
```
Comment (aneelac22) [inline src/helpers/service-account/service-account-helper.ts:31]: This is kind of confusing to have clientID and clientId in the same payload, everything else looks good

**Review by aneelac22 — APPROVED** [review]

**Review by aneelac22 — COMMENTED** [review]
The snapshot tests need to be updated

File: src/helpers/service-account/service-account-helper.ts (hunk 1 above)
Comment (fhlavac) [reply src/helpers/service-account/service-account-helper.ts:31]: It is, but the RBAC API currently retrieves us ServiceAccounts with `clientID`  and https://sso.stage.redhat.com/auth/realms/redhat-external/apis/service_accounts gives us `clientId`. I think it is better than introducing another interface for the same thing. I'll talk to backend folks to get rid of it

**Comment by fhlavac** [comment]
> The snapshot tests need to be updated

Sorry, forgot to push them, thank you 🙂

**Comment by aneelac22** [comment]
build fails due to quay being private, we can retest and merge when that is resolved

**Pull request merged**